workflows/gas_nomenclature.nf

```diff
@@ -72,29 +72,18 @@ workflow GAS_NOMENCLATURE {
     input = Channel.fromSamplesheet("input")
 
     // Ensure meta.id and mlst_file keys match; generate error report for samples where id ≠ key
-    id_key = INPUT_CHECK(input)
+    id_key = INPUT_ASSURE(input)
     ch_versions = ch_versions.mix(id_key.versions)
 
     // Update metadata to include the id_key.match data
     match = id_key.match.map { meta, file, json ->
         def id_match = file.text.trim()
         [meta + [id_match: id_match == 'True'], json]
     }
-
-    // If samples have a disparity between meta.id and JSON key: Exclude the queried samples OR halt the pipeline with an error if sample has an associated cluster address (reference)
-    new_input = match.filter { meta, json ->
-        if (meta.id_match) {
-            return true // Keep the sample
-        } else if (meta.address == null && !meta.id_match) {
-            return false // Remove the sample
-        } else if (meta.address != null && !meta.id_match) {
-            // Exit with error statement
-            throw new RuntimeException("Pipeline exiting: sample with ID ${meta.id} does not have matching MLST JSON file.")
-        }
-    }
+    match.view()
 
     // Prepare reference and query TSV files for LOCIDEX_MERGE
-    profiles = new_input.branch{
+    profiles = match.branch {
         query: !it[0].address
     }
```

Comment:
Is this for only querying profiles that have don't already have an address?

Reply:
Yes exactly 😄